crates/paths/src/paths.rs

```diff
@@ -5,61 +5,84 @@ use std::sync::OnceLock;
 
 pub use util::paths::home_dir;
 
+/// A default editorconfig file name to use when resolving project settings.
+pub const EDITORCONFIG_NAME: &str = ".editorconfig";
+
+// Custom data directory override, set only by set_custom_data_dir.
+static CUSTOM_DATA_DIR: OnceLock<PathBuf> = OnceLock::new();
+
+// Resolved data directory, combining custom or platform defaults, set once.
+static CURRENT_DATA_DIR: OnceLock<PathBuf> = OnceLock::new();
```

Comment:
Why do we need this third static? I would have thought the check would be something like:

```rs
pub fn set_custom_data_dir(dir: &str) -> &'static PathBuf {
  if CUSTOM_DATA_DIR.get().is_some() {
   panic!();
  } 
  // Set custom data dir
}
```

And otherwise, it would be the same as it was earlier in the PR

Reply:
I added `CURRENT_DATA_DIR` and `CONFIG_DIR` as separate statics to ensure `set_custom_data_dir` can only be called before any directory (data or config) is initialized. Checking only `CUSTOM_DATA_DIR.get().is_some()` would allow setting a custom directory after `data_dir()` or `config_dir()` has been resolved, which could lead to inconsistent paths during runtime. If you are okay with that possibility, I can simplify it as suggested - let me know what you think!